### PR TITLE
[SPARK-18016][SQL][FOLLOW-UP] Code Generation: Constant Pool Limit - reduce entries for mutable state

### PR DESCRIPTION
**Pull request by kiszk**
## What changes were proposed in this pull request?

This PR addresses additional review comments in #19811

## How was this patch tested?

Existing test suites

### CODE REVIEW

**Comment by kiszk** [comment]
Jenkins, retest this please

**Comment by kiszk** [comment]
Jenkins, retest this please

**Comment by HyukjinKwon** [comment]
retest this please

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala [hunk 1]
```diff
@@ -118,9 +118,8 @@ case class Like(left: Expression, right: Expression) extends StringRegexExpressi
       if (rVal != null) {
         val regexStr =
           StringEscapeUtils.escapeJava(escape(rVal.asInstanceOf[UTF8String].toString()))
-        // inline mutable state since not many Like operations in a task
         val pattern = ctx.addMutableState(patternClass, "patternLike",
-          v => s"""$v = ${patternClass}.compile("$regexStr");""", forceInline = true)
+          v => s"""$v = ${patternClass}.compile("$regexStr");""")
```
Comment (ueshin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala]: nit: we can remove `{` and `}` around `patternClass`.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala [hunk 2]
```diff
@@ -194,9 +193,8 @@ case class RLike(left: Expression, right: Expression) extends StringRegexExpress
       if (rVal != null) {
         val regexStr =
           StringEscapeUtils.escapeJava(rVal.asInstanceOf[UTF8String].toString())
-        // inline mutable state since not many RLike operations in a task
         val pattern = ctx.addMutableState(patternClass, "patternRLike",
-          v => s"""$v = ${patternClass}.compile("$regexStr");""", forceInline = true)
+          v => s"""$v = ${patternClass}.compile("$regexStr");""")
```
Comment (ueshin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala]: ditto.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala (hunk 1 above)
Comment (kiszk) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala]: Sure, done for other places, too.

**Comment by kiszk** [comment]
retest this please

**Comment by kiszk** [comment]
retest this please

**Comment by kiszk** [comment]
ping @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala [hunk 3]
```diff
@@ -283,7 +283,7 @@ case class InputAdapter(child: SparkPlan) extends UnaryExecNode with CodegenSupp
 
   override def doProduce(ctx: CodegenContext): String = {
     // Right now, InputAdapter is only used when there is one input RDD.
-    // inline mutable state since an inputAdaptor in a task
+    // inline mutable state since an InputAdapter in a task
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala]: you miss some words...

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala [hunk 4]
```diff
@@ -299,7 +299,7 @@ class CodegenContext {
   def initMutableStates(): String = {
     // It's possible that we add same mutable state twice, e.g. the `mergeExpressions` in
     // `TypedAggregateExpression`, we should call `distinct` here to remove the duplicated ones.
-    val initCodes = mutableStateInitCode.distinct
+    val initCodes = mutableStateInitCode.distinct.map(_ + "\n")
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala]: ah, good catch!

**Comment by cloud-fan** [comment]
LGTM

File: sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala (hunk 3 above)
Comment (kiszk) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala]: sure, done

File: sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala [hunk 5]
```diff
@@ -283,7 +283,7 @@ case class InputAdapter(child: SparkPlan) extends UnaryExecNode with CodegenSupp
 
   override def doProduce(ctx: CodegenContext): String = {
     // Right now, InputAdapter is only used when there is one input RDD.
-    // inline mutable state since an inputAdaptor in a task
+    // inline mutable state since an InputAdapter is used once in a task for WholeStageCodegen
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala]: nit: `inline` -> `Inline`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala [hunk 6]
```diff
@@ -440,8 +440,9 @@ case class SortMergeJoinExec(
     val spillThreshold = getSpillThreshold
     val inMemoryThreshold = getInMemoryThreshold
 
+    // inline mutable state since not many join operations in a task
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala]: nit: `Inline`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala [hunk 7]
```diff
@@ -587,20 +587,24 @@ case class HashAggregateExec(
...
-        ctx.addMutableState(s"java.util.Iterator<InternalRow>", "vectorizedFastHashMapIter")
+          v => s"$v = new $fastHashMapClassName();", forceInline = true)
+        ctx.addMutableState(s"java.util.Iterator<InternalRow>", "vectorizedFastHashMapIter",
+          forceInline = true)
       } else {
         val generatedMap = new RowBasedHashMapGenerator(ctx, aggregateExpressions,
           fastHashMapClassName, groupingKeySchema, bufferSchema).generate()
         ctx.addInnerClass(generatedMap)
 
+        // inline mutable state since not many aggregation operations in a task
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala]: nit: `Inline`

File: sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala [hunk 8]
```diff
@@ -587,20 +587,24 @@ case class HashAggregateExec(
...
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/HashAggregateExec.scala]: nit: `Inline`

**Comment by viirya** [comment]
LGTM

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**